### PR TITLE
fix audio client starve detection

### PR DESCRIPTION
**Pull request by zzmp**

### CODE REVIEW

**Comment by zzmp** [comment]
Fixes an issue where audio starves on the output buffer were not detected by the audio client.

To test, I've added a commit (`DNM: test to starve audio client`), and applied it to master and to d7643ef, in #8585 and #8586, respectively.
- Verify that the test commits are the same in these PRs.
- Verify that the commit hash is the same in #8584 and #8586 (d7643ef).
- Run #8585's Interface. This will starve your audio output buffer, but not be detected. You should hear stuttering.
- Run #8586's Interface. This changes the detection criteria. You should not hear stuttering.
- Run #8584 as a sanity check. It should work fine. Audio smoke test.

**Comment by hifi-gustavo** [comment]
Build console output for commit: d7643ef2a0a4b8b7cbe0dcd23cc78b622765a8ce available at: https://gist.github.com/be670b30661ff50f118ccf323840453f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8584-d7643ef.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8584-d7643ef.dmg

File: libraries/audio-client/src/AudioClient.cpp [hunk 1]
```diff
@@ -1383,24 +1386,20 @@ qint64 AudioClient::AudioOutputIODevice::readData(char * data, qint64 maxSize) {
...
-        // return 0s
+        // nothing on network, don't grab anything from injectors, and just return 0s
+        // this will flood the log: qCDebug(audioclient, "empty/partial network buffer");
         memset(data, 0, maxSize);
         bytesWritten = maxSize;
-        
     }
 
     int bytesAudioOutputUnplayed = _audio->_audioOutput->bufferSize() - _audio->_audioOutput->bytesFree();
     if (!bytesAudioOutputUnplayed) {
```
Comment (ZappoMan) [inline libraries/audio-client/src/AudioClient.cpp]: couple of things...

1) The coding standard clearly says to not test non-bool (or pointers) as if they were bools... so if you are testing for `bytesAudioOutputUnplayed == 0` or `bytesAudioOutputUnplayed != 0` then explicitly spell that out... I realize you didn't write this code, but you're touching it, so you need to fix it.

2) maybe we should still include the debug output, but just make it a suppressible output so it doesn't flood the log

Comment (zzmp) [inline libraries/audio-client/src/AudioClient.cpp]: 1) will fix shortly
2) this is redundant, as it is also logged in outputNotify, as the number of unfulfilled reads, so I do not want to include this in the log here.

Comment (ZappoMan) [inline libraries/audio-client/src/AudioClient.cpp]: re 2: ok

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2a2673d8980260544823ba7802f4c4d9a565e7af available at: https://gist.github.com/be670b30661ff50f118ccf323840453f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8584-2a2673d.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8584-2a2673d.dmg

**Comment by OmegaHeron** [comment]
Testing Windows 10
Sanity tests were looked at before.
This specific PR
1) I hear sounds from audio injectors properly in standard home scene and dev-playa
2) Server audio loopback is clean in all tested locations (dev-playa, heron and my LAN server) covering a wide range of client->server ping times.
3) This works proper with my bluetooth headset, usb audio device and mobo device.

QA Passs.

**Pull request merged**